src/hdnode.js

```diff
@@ -64,7 +64,9 @@ HDNode.fromBase58 = function (string, networks) {
   if (Array.isArray(networks)) {
     network = networks.filter(function (x) {
       return version === x.bip32.private ||
-             version === x.bip32.public
+             version === x.bip32.public ||
+             version === x.bip49.private ||
```

Comment:
I think these semantics should be opt-in. Imagine someone supported Bitcoin Cash, and imported keys. There's not currently a bitcoin-cash network in the library, so they're using the bitcoin network for that. Imagine they didn't read the release notes, and allowed users to import a P2SH(P2WPKH) address, and funded that on Bitcoin Cash.. They're unspendable without talking to a miner.

I think this feature should be opt in, so a new function `bip44TypefromBase58` or something that describes what the function will do (parse a HD key with an embedded script type, constructing a hdnode which produces the right address instead of always P2PKH)

Reply:
@afk11 I don't understand,  what about these new BIP49 constants would trap users into creating unspendable addresses?

Reply:
Isn't the risk that any P2SH(P2WPKH) scripts are unspendable by BCASH nodes?